### PR TITLE
Gherkin display task runs associated with pipeline run odc 3712

### PR DESCRIPTION
**Pull request by sanketpathak**
Epic: https://issues.redhat.com/browse/ODC-3712
Story: https://issues.redhat.com/browse/ODC-4851

### CODE REVIEW

**Review by gajanan-more — COMMENTED** [review]
You haven't added the file extension

File: frontend/packages/dev-console/integration-tests/features/pipelines/pipeline-Display_task_runs_ODC-3712.feature [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    Given user is at pipeline details page
+    And pipeline runs are present
+    When user clicks on pipeline runs tab
+    And user clicks on a pipeline run
+    And user clicks on Task Runs tab
+    And user clicks on a task run
+    Then user is redirected to Task Run Details tab
+    And user can see "Details", "Log", "YAML" and "Events" tab
+    And user can see Status and Log snippet in "Details" tab
+
```
Comment (gajanan-more) [inline frontend/packages/dev-console/integration-tests/features/pipelines/pipeline-Display_task_runs_ODC-3712.feature:41]: Remove this extra line

File: frontend/packages/dev-console/integration-tests/features/pipelines/pipeline-Display_task_runs_ODC-3712.feature [hunk 2]
```diff
@@ -0,0 +1,41 @@
+Feature: Display task runs page
+    As a developer, I would like to navigate from a PipelineRun to the related TaskRuns and their associated pods so that I can debug my pipelines by looking at the volumes, secrets, etc that are made available to the taskruns and pods.
+
+Background:
+    Given user is at developer perspective
+    And user has installed Openshift Pipelines Operator
```
Comment (gajanan-more) [inline frontend/packages/dev-console/integration-tests/features/pipelines/pipeline-Display_task_runs_ODC-3712.feature:6]:
```suggestion
    And user has installed OpenShift Pipelines Operator
```

**Review by gajanan-more — CHANGES_REQUESTED** [review]

File: frontend/packages/dev-console/integration-tests/features/pipelines/pipeline-Display_task_runs_ODC-3712.feature (hunk 2 above)
Comment (sanketpathak) [reply frontend/packages/dev-console/integration-tests/features/pipelines/pipeline-Display_task_runs_ODC-3712.feature]: Updated

File: frontend/packages/dev-console/integration-tests/features/pipelines/pipeline-Display_task_runs_ODC-3712.feature (hunk 1 above)
Comment (sanketpathak) [reply frontend/packages/dev-console/integration-tests/features/pipelines/pipeline-Display_task_runs_ODC-3712.feature]: Updated

**Review by gajanan-more — APPROVED** [review]

File: frontend/packages/dev-console/integration-tests/features/pipelines/pipeline-Display_task_runs_ODC-3712.feature [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+Scenario: Task Runs Details page
+    Given user is at pipeline details page
+    And pipeline runs are present
+    When user clicks on pipeline runs tab
+    And user clicks on a pipeline run
+    And user clicks on Task Runs tab
+    And user clicks on a task run
+    Then user is redirected to Task Run Details tab
+    And user can see "Details", "Log", "YAML" and "Events" tab
+    And user can see Status and Log snippet in "Details" tab
```
Comment (karthikjeeyar) [inline frontend/packages/dev-console/integration-tests/features/pipelines/pipeline-Display_task_runs_ODC-3712.feature:40]: Break into two scenarios

1. For a succeeded task run only `status` and `pod` fields are shown
image
2. For a failed task run, following fields are shown
image

File: frontend/packages/dev-console/integration-tests/features/pipelines/pipeline-Display_task_runs_ODC-3712.feature [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+    Then user can see different task runs based on number of tasks present in pipeline
+    And user can see "Name", "Task", "Pod", "Status" and "Started" columns
+
+
+@regression
+Scenario: Options in breadcrumbs of task runs
+    Given user has pipeline runs present
+    And user is on Pipeline Runs Details page
+    When user clicks on Task Runs tab
+    And user clicks on breadcrumb of a task run
```
[the remark this answers is not in the document]
Comment (sanketpathak) [reply frontend/packages/dev-console/integration-tests/features/pipelines/pipeline-Display_task_runs_ODC-3712.feature]: Updated

File: frontend/packages/dev-console/integration-tests/features/pipelines/pipeline-Display_task_runs_ODC-3712.feature [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+
+
+@regression
+Scenario: Task runs tab
+    Given user is at pipeline details page
+    And pipeline runs are present
+    When user clicks on pipeline runs tab
+    And user clicks on a pipeline run to see Task Runs tab
+    And user clicks on Task Runs tab
+    Then user can see different task runs based on number of tasks present in pipeline
```
[the remark this answers is not in the document]
Comment (sanketpathak) [reply frontend/packages/dev-console/integration-tests/features/pipelines/pipeline-Display_task_runs_ODC-3712.feature]: Updated

File: frontend/packages/dev-console/integration-tests/features/pipelines/pipeline-Display_task_runs_ODC-3712.feature (hunk 3 above)
Comment (sanketpathak) [reply frontend/packages/dev-console/integration-tests/features/pipelines/pipeline-Display_task_runs_ODC-3712.feature]: Added two scenarios for passed and failed task runs

File: frontend/packages/dev-console/integration-tests/features/pipelines/pipeline-Display_task_runs_ODC-3712.feature [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+    And user clicks on a pipeline run to see Task Runs tab
+    And user clicks on Task Runs tab
+    Then user can see different task runs based on number of tasks executed in pipeline
+    And user can see "Name", "Task", "Pod", "Status" and "Started" columns
+
+
+@regression
+Scenario: Options in kebab menu of task runs
+    Given user has pipeline runs present
+    And user is on Pipeline Runs Details page
```
Comment (makambalaji) [inline frontend/packages/dev-console/integration-tests/features/pipelines/pipeline-Display_task_runs_ODC-3712.feature:24]:
```suggestion
    Given user is at pipeline details page with pipeline runs
```

File: frontend/packages/dev-console/integration-tests/features/pipelines/pipeline-Display_task_runs_ODC-3712.feature [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+    When user clicks on Task Runs tab
+    And user clicks on a kebab menu of a task run in the list
+    Then user can see "Edit labels", "Edit annotations", "Edit Task Run" and "Delete Task Run"
```
Comment (makambalaji) [inline frontend/packages/dev-console/integration-tests/features/pipelines/pipeline-Display_task_runs_ODC-3712.feature:27]:
```suggestion
    Then user can see kebab menu options Edit labels, Edit annotations, Edit Task Run and Delete Task Run
```

File: frontend/packages/dev-console/integration-tests/features/pipelines/pipeline-Display_task_runs_ODC-3712.feature [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+
+
+@regression
+Scenario: Task Runs Details page for passed task runs
+    Given user is at pipeline details page
+    And pipeline runs are present
```
Comment (makambalaji) [inline frontend/packages/dev-console/integration-tests/features/pipelines/pipeline-Display_task_runs_ODC-3712.feature:33]:
```suggestion
       Given user is at pipeline details page with pipeline runs
```

File: frontend/packages/dev-console/integration-tests/features/pipelines/pipeline-Display_task_runs_ODC-3712.feature [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+    Then user can see different task runs based on number of tasks executed in pipeline
+    And user can see "Name", "Task", "Pod", "Status" and "Started" columns
+
+
+@regression
+Scenario: Options in kebab menu of task runs
+    Given user has pipeline runs present
+    And user is on Pipeline Runs Details page
+    When user clicks on Task Runs tab
+    And user clicks on a kebab menu of a task run in the list
```
Comment (makambalaji) [inline frontend/packages/dev-console/integration-tests/features/pipelines/pipeline-Display_task_runs_ODC-3712.feature:26]:
```suggestion
    And user clicks kebab menu of a task run
```

File: frontend/packages/dev-console/integration-tests/features/pipelines/pipeline-Display_task_runs_ODC-3712.feature [hunk 10]
```diff
@@ -0,0 +1,53 @@
...
+    And user clicks on a Succeeded task run
+    Then user is redirected to Task Run Details tab
+    And user can see "Details", "Log", "YAML" and "Events" tab
+    And user can see Status and Pods in "Details" tab
+
+
+@regression
+Scenario: Task Runs Details page for failed task runs
+    Given user is at pipeline details page
+    And pipeline runs are present
```
Comment (makambalaji) [inline frontend/packages/dev-console/integration-tests/features/pipelines/pipeline-Display_task_runs_ODC-3712.feature:46]:
```suggestion
     Given user is at pipeline details page with pipeline runs
```

**Review by makambalaji — COMMENTED** [review]
PTAL the comments

File: frontend/packages/dev-console/integration-tests/features/pipelines/pipeline-Display_task_runs_ODC-3712.feature [hunk 11]
```diff
@@ -0,0 +1,53 @@
...
+Background:
+    Given user is at developer perspective
+    And user has installed OpenShift Pipelines Operator
+    And user has selected "aut-pipeline-taskruns" namespace
+
+
+@regression
+Scenario: Task runs tab
+    Given user is at pipeline details page
+    And pipeline runs are present
```
[the remark this answers is not in the document]
Comment (sanketpathak) [reply frontend/packages/dev-console/integration-tests/features/pipelines/pipeline-Display_task_runs_ODC-3712.feature]: Updated

File: frontend/packages/dev-console/integration-tests/features/pipelines/pipeline-Display_task_runs_ODC-3712.feature (hunk 6 above)
Comment (sanketpathak) [reply frontend/packages/dev-console/integration-tests/features/pipelines/pipeline-Display_task_runs_ODC-3712.feature]: Updated

File: frontend/packages/dev-console/integration-tests/features/pipelines/pipeline-Display_task_runs_ODC-3712.feature (hunk 7 above)
Comment (sanketpathak) [reply frontend/packages/dev-console/integration-tests/features/pipelines/pipeline-Display_task_runs_ODC-3712.feature]: Updated

File: frontend/packages/dev-console/integration-tests/features/pipelines/pipeline-Display_task_runs_ODC-3712.feature (hunk 8 above)
Comment (sanketpathak) [reply frontend/packages/dev-console/integration-tests/features/pipelines/pipeline-Display_task_runs_ODC-3712.feature]: Updated

File: frontend/packages/dev-console/integration-tests/features/pipelines/pipeline-Display_task_runs_ODC-3712.feature (hunk 9 above)
Comment (sanketpathak) [reply frontend/packages/dev-console/integration-tests/features/pipelines/pipeline-Display_task_runs_ODC-3712.feature]: Updated

File: frontend/packages/dev-console/integration-tests/features/pipelines/pipeline-Display_task_runs_ODC-3712.feature (hunk 10 above)
Comment (sanketpathak) [reply frontend/packages/dev-console/integration-tests/features/pipelines/pipeline-Display_task_runs_ODC-3712.feature]: Updated

**Pull request merged**